src/components/Table/index.tsx

```diff
@@ -212,11 +124,15 @@ export default class Table<T> extends React.Component<
 			);
 		}
 
+		const sortState = this.props.defaultSort || {
```

Comment:
This feels a bit weird, can you not just make the sort controllable through a `sort` property? This `defaultSort` property is limited in functionality.
If you maintained the same constructor logic you could set the state on update with a `componentWillUpdate` lifecycle method, eg:
```js
componentWillUpdate(nextProps) {
  if (!isEqual(nextProps.sort, this.props.sort)) {
    this.setState({ sort: nextProps.sort })
  }
}
```

Reply:
It is actually not recommended to do that (and componentWillUpdate is depricated in following versions of React): https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#preferred-solutions. I do agree that it gives more control like that, so I am up for doing it as it will be easier to change the behavior later on without breaking the API.

Reply:
Reading that link it appears that the recommendation is to use a completely controlled component, or use `getDerivedStateFromProps` right? They don't seem to explore the idea of *optionally* controlled components, but it seems reasonable that `getDerivedStateFromProps` should do what we are looking for

Reply:
`getDerivedStateFromProps` does allow us to do what we want, they just recommend to avoid it if possible as it can create race conditions with internal state management and give you a result you don't expect (as described in their example). For now we can use `componentWillReceiveProps` and once we upgrade to React 16.3+ we can refactor away from the depricated lifecycle methods.